app/src/main/java/es/wolfi/app/passman/autofill/CredentialAutofillService.java

```diff
@@ -115,7 +116,7 @@ public void onFillRequest(FillRequest request, CancellationSignal cancellationSi
         CredentialAutofillService.WebDomainResult domain = getLikelyDomain(structures);
 
         // Grab Credentials from vault
-        ArrayList<Credential> allCred = v.getCredentials();
+        CopyOnWriteArrayList<Credential> allCred = v.getCredentials();
```

Comment:
Won't be better to use synchronized on the asynchronous task instead of usin this super heavy type?

Reply:
could not solve the problem with
```
synchronized (credentials) {
...
}
```